backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -55,17 +55,17 @@ impl JsonParser {
     /// A boolean if the image has an valid license or not.
     /// If the image has no license or no license history, the image isn't restricted by any license and true will be returned.
     #[must_use]
-    pub fn check_license(root: &JsonRootLicense) -> bool {
+    pub fn check_license(root: &JsonRootLicense) -> (bool, String) {
```

Comment:
a more idiomatic return type would be `Result<(), String>` or ever better produce the error here.